tools/slang-unit-test/unit-test-math.cpp

```
@@ -0,0 +1,332 @@
+    // Large magnitudes — relative tolerance applies.
+    SLANG_CHECK(Math::AreNearlyEqual(1e20, 1e20 + 100, eps));
+    // But not when the gap exceeds eps * (|a|+|b|).
+    SLANG_CHECK(!Math::AreNearlyEqual(1.0, 2.0, eps));
+}
+
+SLANG_UNIT_TEST(mathAreNearlyEqualNearZero)
+{
+    // Near-zero values use absolute tolerance to avoid the
+    // relative-comparison singularity.
+    SLANG_CHECK(Math::AreNearlyEqual(0.0, 0.0, 1e-9));
+    // Different signs but tiny magnitudes — still equal-ish.
```

Cppcheck (2.20.0)

[style] The function 'fromLiteral' is never used.

(unusedFunction)